xarray_pickler/xarray_pickler.py

```diff
@@ -13,48 +13,90 @@
+                "w" will join the writeable pickle directory from CONFIG and the base pickle path if a writeable pickle directory has been supplied. Otherwise None is returned.
+
+    :return: The read or write pickle path.
+    """
 
-    if not os.path.isdir(pickle_path):
-        os.makedirs(pickle_path)
+    pickle_dirs = CONFIG["paths"]["pickle_dirs"]
+    writeable_pickle_dir = CONFIG["paths"]["writeable_pickle_dir"]
 
-    return pickle_path + ".pickle"
+    if mode == "r":
+        for pdir in pickle_dirs:
+
+            pickle_path = os.path.join(pdir, pck.lstrip("/"))
+            if os.path.isfile(pickle_path):
+                # found a pickle file so can return the path
+                return pickle_path
+
+    if mode == "w" and writeable_pickle_dir:
+        pickle_path = os.path.join(writeable_pickle_dir, pck.lstrip("/"))
+
+        os.makedirs(os.path.dirname(pickle_path), exist_ok=True)
+
+        return pickle_path
+
+    return
 
 
 def open_dset(dpath, force_repickle=False, **kwargs):
     """
     Open xarray.Dataset object. If previously pickled, it will be opened from the pickle file stored in the cache.
-    Otherwise, it will be pickled and stored in the cache and opened using xarray.open_mfdataset() with any extra keyword arguments specified.
+    Otherwise, it will be pickled and stored in the cache, if a cache is specified, after it is opened using xarray.open_mfdataset() with any extra keyword arguments specified.
+    If there is no writeable pickle directory specified in the config, and the pickle does not already exist, the dataset will just be opened using xarray and returned.
 
     :param dpath (str): Directory path to netCDF files to generate dataset from e.g. "/badc/cmip6/data/CMIP6/CMIP/IPSL/IPSL-CM6A-LR/historical/r1i1p1f1/Amon/rlds/gr/v20180803"
-    :param force_repickle: If True, the xarray.Dataset object will be repickled. Default is False.
+    :param force_repickle: If True, the xarray.Dataset object will be repickled, if a writeable pickle directory is specified in the config. Default is False.
     :param **kwargs: Other keyword arguments that can be used in xarray.open_mfdataset(). Used only the first time a dataset is pickled or if force_repickle=True.
 
     :return: xarray.Dataset object
     """
-
     open_kwargs = CONFIG["open_mfdataset_kwargs"].copy()
     open_kwargs.update(kwargs)
 
+    if dpath.endswith("/"):
+        dpath = dpath[:-1]
+
     fpattn = f"{dpath}/*.nc"
     logger.info(f"Reading: {fpattn}")
 
-    pickle_path = _get_pickle_path(dpath)
+    pck = _get_pickle_name(dpath)
+    read_pickle_path = _get_pickle_path(pck, mode="r")
 
-    if os.path.isfile(pickle_path) and not force_repickle:
+    if read_pickle_path and not force_repickle:
         try:
-            with open(pickle_path, "rb") as reader:
+            with open(read_pickle_path, "rb") as reader:
                 return pickle.load(reader)
```

Comment:
It would be useful to add a logger message here to explain that you have read the Dataset from the pickle file:

```
            with open(read_pickle_path, "rb") as reader:
                ds = pickle.load(reader)
                logger.info(f"Dataset read from cached pickle: {read_pickle_path}")
                return ds
```